triton-client/src/components/DatasetColumns.tsx

```diff
@@ -0,0 +1,371 @@
+        columns.push({
+            title: "",
+            dataIndex: "id",
+            key: "metric",
+            width: "1%",
+            align: "center",
+            render: (id, dataset) => <MetricButton dataset={dataset} />,
+        })
+        columns.push({
+            title: "Dataset ID",
+            dataIndex: "id",
+            key: "id",
+            width: "5%",
+            sorter: (a, b) => a.id - b.id,
+        })
+        columns.push({
+            title: "Lane",
+            dataIndex: "lane",
+            key: "lane",
+            width: "5%",
+            sorter: (a, b) => a.lane - b.lane,
+            render: (lane) => <>{`Lane ${lane}`}</>,
+        })
+        columns.push({
+            title: "SFTP",
+            dataIndex: "id",
+            key: "sftp",
+            width: "5%",
+            render: (id, dataset) =>
+                dataset.isFetchingRequest ? (
+                    <Spin />
+                ) : (
+                    <StagingButton
+                        dataset={dataset}
+                        type={"SFTP"}
+                        updateDataset={updateDataset}
+                    />
+                ),
+        })
+        columns.push({
+            title: "Globus",
+            dataIndex: "id",
+            key: "globus",
+            width: "5%",
+            render: (id, dataset) =>
+                dataset.isFetchingRequest ? (
+                    <Spin />
+                ) : (
+                    <StagingButton
+                        dataset={dataset}
+                        type={"GLOBUS"}
+                        updateDataset={updateDataset}
+                    />
+                ),
+        })
+        columns.push({
+            title: "Expiration",
+            dataIndex: "id",
+            key: "expiration",
+            width: "10%",
+            render: (id, { activeRequest }) => (
+                <Expiration activeRequest={activeRequest} />
+            ),
+        })
+        columns.push({
+            title: "Latest Release Date (UTC)",
+            dataIndex: "latest_release_update",
+            key: "latest_release_date",
+            width: "10%",
+            defaultSortOrder: "descend",
+            sorter: (a, b) =>
+                a.latest_release_update.getTime() -
+                b.latest_release_update.getTime(),
+            render: (date) => {
+                if (date) {
+                    return date.toUTCString()
+                } else {
+                    return "-"
+                }
+            },
+        })
+        columns.push({
+            title: "Size",
+            dataIndex: "id",
+            key: "size",
+            width: "5%",
+            render: (id, { totalSize }) =>
+                totalSize !== 0 ? (
+                    <>{dataSize(totalSize).join(" ")}</>
+                ) : (
+                    <Spin />
+                ),
+        })
+        return columns
+    }, [])
+}
+
+interface MetricProps {
```

Comment:
there are multiple interfaces in this file. i suggest to refactor this file to be more modular, as per my previous comment.